sympy/matrices/matrices.py

```diff
@@ -1935,6 +1935,15 @@ def __str__(self):
             return 'Matrix(%s, %s, [])' % (self.rows, self.cols)
         return "Matrix(%s)" % str(self.tolist())
 
+    def __mod__(self, other):
```

Comment:
This should be `_eval_Mod` and it should be in the class MatrixArithmetics, along with `_eval_Abs`. 

See https://github.com/sympy/sympy/blob/master/sympy/matrices/common.py#L1895  which shows how to return a matrix of the same class, and without running an explicit loop like this:  
```
    def _eval_Mod(self):
        return self._new(self.rows, self.cols, lambda i, j: Mod(self[i, j]))
```

Unless you know you are working with numbers, Mod is to be preferred to `%`, which may or may not be implemented by a SymPy object that a matrix contains. For example, Matrix entries may be Polynomials, in which case `%` does not work as one would expect.  If Mod does not know how to evaluate the mod of something, it will return `Mod(..,, ...)` instead of giving a wrong/nonsensical result that % might.

Reply:
Thanks for the comments. I'll work on this